### PR TITLE
executor: migrate test-infra to testify for parallel_apply_test.go

### LINKED ISSUE #28599: migrate test-infra to testify for `parallel_apply_test.go`

### PR DESCRIPTION
**Pull request by feitian124**
### What problem does this PR solve?

Issue Number: close #28599 

### Release note

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

**Review by xiongjiwei — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]

File: executor/parallel_apply_test.go [hunk 1]
```diff
@@ -38,35 +39,39 @@ func checkApplyPlan(c *C, tk *testkit.TestKit, sql string, parallel int) {
...
-	c.Assert(containApply, IsTrue)
+	require.True(t, containApply)
 }
 
-func (s *testSuite) TestParallelApply(c *C) {
-	tk := testkit.NewTestKitWithInit(c, s.store)
+func TestParallelApplyPlan(t *testing.T) {
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	tk := testkit.NewTestKitWithInit(t, store)
```
Comment (tisonkun) [inline executor/parallel_apply_test.go:54]:
```suggestion
	tk := testkit.NewTestKitWithInit(t, store)
	tk.MustExec("use test")
```

ditto others

File: testkit/testkit.go [hunk 2]
```diff
@@ -73,6 +73,13 @@ func NewTestKitWithSession(t testing.TB, store kv.Storage, se session.Session) *
 	}
 }
 
+// NewTestKitWithInit returns a new *TestKit and use test db.
+func NewTestKitWithInit(t testing.TB, store kv.Storage) *TestKit {
+	tk := NewTestKit(t, store)
+	tk.MustExec("use test")
+	return tk
+}
```
Comment (tisonkun) [inline testkit/testkit.go:81]: Please remove this method.

**Review by tisonkun — CHANGES_REQUESTED** [review]
`NewTestKitWithInit` is somehow too much to encapsulate and I intend to avoid it but with one more line `tk.MustExec("use test")`

Most developers aren't aware difference between `NewTestKitWithInit` and `NewTestKit` but choose either randomly. I think make it explicit should help.

File: executor/parallel_apply_test.go (hunk 1 above)
Comment (tisonkun) [reply executor/parallel_apply_test.go:54]: I'm going to push a patch now.

**Review by tisonkun — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**